pkg/composer/kill.go

```diff
@@ -0,0 +1,45 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package composer
+
+import (
+	"context"
+
+	"github.com/sirupsen/logrus"
+)
+
+type KillOptions struct {
+	Signal string
+}
+
+func (c *Composer) Kill(ctx context.Context, opts KillOptions, services []string) error {
+	serviceNames, err := c.ServiceNames(services...)
+	if err != nil {
+		return err
+	}
+	containers, err := c.Containers(ctx, serviceNames...)
+	if err != nil {
+		return err
+	}
+	for _, container := range containers {
+		args := []string{"kill", "-s", opts.Signal, container.ID()}
+		if err := c.runNerdctlCmd(ctx, args...); err != nil {
+			logrus.Warn(err)
+		}
+	}
+	return nil
```

Comment:
Shouldn’t we return an error if failed to kill at least one of the containers?

Reply:
Or return the first error?